README.md

```diff
@@ -651,29 +651,59 @@ separate section above).
 *form file contracts before running the deploy script. Otherwise the health
 *checks will fail as your node is not ready to upload and download
 
-### Run Docker Command
+### Run Shell Commands
 
-Playbook:
+There are several options to run shell commands against portals via Ansible.
 
-- Run a docker command on portals define in `docker_commands` variable.
+Requires (all options):
 
-Preparation:  
-Defined a `docker_commands` variable in your `portal_versions.yml` file.
+- `shell_commands` variable must be defined as a list of commands to be run.
+- It can be defined either in `my-vars/portal_versions.yml` (e.g. for
+  `portals-deploy` playbook with other variables) or in a separate file e.g.
+  `my-vars/shell-commands.yml`
+
+Depreciation note:
+
+Previously there was `docker_commands` variable with the similar functionality,
+its usage is now depreciated, but for backwards compatibility it works as it
+used to work.
+
+If both `docker_commands` and `shell_commands` are used, commands in both of
+these lists are run (commands from these two lists are added together).
 
 Example:
 
 ```yaml
 ---
-docker_commands:
+shell_commands:
   - "docker exec sia siac"
   - "docker exec sia siac renter"
 ```
 
+Wrapping commands into quotes is optional, but might be needed if Ansible/Yaml
+syntax requires it.
+
+#### Disable from a load balancer, run shell commands, test and enable portals
+
+See above for `shell_commands` configuration.
+
+Playbook:
+
+- Disables health check.
+- Runs shell commands defined in `shell_commands` variable and logs outputs.
+- Runs integration tests and health checks.
+- Enables portals.
```

Comment:
How do we run a command without disabling health checks?

It is surprising to me that running a shell command will pull a sever from the LB. That seems to be a responsibility which should lie with the operator - they are aware of what type of command they are running and whether that requires the server to be pulled first.

I suggest we make that configurable via a parameter and switch it off by default.

Reply:
> It is surprising to me that running a shell command will pull a sever from the LB. That seems to be a responsibility which should lie with the operator - they are aware of what type of command they are running and whether that requires the server to be pulled first.

I think that makes sense when doing things manually, but i disagree with ansible playbooks.  The pattern we have enforced is that playbooks handle everything needed safely. Making developers have to know when to manually do things along side of playbooks is an anti-pattern IMO.

Reply:
Something I've used this playbook for multiple times already is to kick off Pinner sweeps on large groups of servers. Another use case I can easily imagine is checking the free disk space on a bunch of servers. Neither of these require servers to be taken out of the LB. Moreover, I would argue that the majority of operations we are ever going to use this playbook for fall in the same category - minor operations, which don't affect the state of the server.

Given the above, having the playbook take servers out of the LB has two drawbacks:
 - a distracted developer can bring down most of the cluster by running a simple `df -h` against it
 - any such operation will take exorbitant amounts of time, despite the only purpose of this playbook being to save us time

If we have the same distracted dev from before and they run something that should bring a server out of the LB and they forget to do that, I would expect the health checks of that server to bring it out the next time they run. That's definitely an undesirable situation but it seems better than a complete cluster meltdown, which would happen if we accidentally remove the majority of the servers and the others die from too much traffic.

**Suggestion**:
In order to make this both usable and safe, I suggest we add a *required* parameter which will define whether the server will be taken out of the LB or not. If the parameter is undefined, the playbook fails to run.